File: Practice(Beginner)/Python3/The_Lead_Game.py
# Source: https://www.codechef.com/problems/TLG

n = int(input())
p1 = []
p2 = []
a,b=map(int,input().split())
p1.append(a)
p2.append(b)
for i in range(1,n):
    a,b = map(int,input().split())
    p1.append(p1[i-1]+a)
    p2.append(p2[i-1]+b)
d = {1:0,2:0}
for i in range(n):
    if(p1[i] > p2[i] and p1[i]-p2[i] > d[1]):
        d[1] = p1[i]-p2[i]
    elif(p2[i] > p1[i] and p2[i]-p1[i] > d[2]):
        d[2] = p2[i]-p1[i]
if(d[1] > d[2]):
    print(1,d[1])
else:
    print(2,d[2])

File: Practice(Beginner)/C++/The Lead Game.cpp
// Source: https://www.codechef.com/problems/TLG

#include <bits/stdc++.h>
using namespace std;

int main() {

    int t, a, b;
    cin >> t;
    vector<int> p1(t,0), p2(t,0);

    unordered_map<int, int> res;
    int maxScoreP1 = INT_MIN;
    int maxScoreP2 = INT_MIN;

    cin >> p1[0] >> p2[0];
    p1[0] > p2[0] ? res[1] = p1[0]-p2[0] : res[2] = p2[0]-p1[0];
    maxScoreP1 = max(maxScoreP1,res[1]);
    maxScoreP2 = max(maxScoreP2,res[2]);

    for (int i = 1; i < t; ++i) {
        cin >> a >> b;

        p1[i] = p1[i-1] + a;
        p2[i] = p2[i-1] + b;

        if(p1[i] > p2[i] ){
            res[1] = p1[i]-p2[i];
            maxScoreP1 = max(maxScoreP1,res[1]);
        } else{
            res[2] = p2[i]-p1[i];
            maxScoreP2 = max(maxScoreP2,res[2]);
        }
    }

    maxScoreP1 > maxScoreP2 ? cout << 1 << " " << maxScoreP1 : cout << 2 << " " << maxScoreP2;

    return 0;
}
